rozwiazania_tesktowe.txt

```diff
@@ -5,6 +5,9 @@ https://drive.google.com/file/d/1FOCU3PVdsIbdJp1pdQPjLzPb-In64F7r/view?usp=shari
 Należałoby zastosować ten typ adapteru w momencie gdy musimy przeładować metody obiektu adaptowanego. W tym przypadku obiektu klasy "DrawPanelController".
 Próbując stworzyć analogię: założmy, że nasz plotter wykonuje rysunki na czerwono. DrawPanelController podczas wywoływania funkcji drawLine stworzyłby jednak linię czarną, a więc symulacja nie pokrywałabym się do końca z tym co w rzeczywistości pojawiłoby się na ploterze. W takim wypadku być może moglibyśmy zechcieć przeciążyć tę metodę drawLine (a w tym celu użyc adaptera klasowego) aby na symulacji linia także pojawiła się jako czerwona.
 
+3.4.3.
+Tak naprawdę takie rozwiązanie jest tak samo daleko od adaptera klasowego jak i obiektowego, ze względu na fakt, że z jednej strony mamy doczynienia z dziedziczeniem, przy jednoczesnym nieimplementowaniu interfejsu klienta.
```

Comment:
Tak właśnie jest zbudowany adapter klasy!

Reply:
https://pl.wikipedia.org/wiki/Adapter_(wzorzec_projektowy)#Struktura
"Różnią się one nieznacznie budową oraz właściwościami. Do stworzenia adaptera klasowego wykorzystywane jest wielokrotne dziedziczenie. Klasa adaptera dziedziczy prywatnie po klasie adaptowanej oraz publicznie implementuje interfejs klienta[1]. W przypadku tego adaptera wywołanie funkcji jest przekierowywane do bazowej klasy adaptowanej[4]."

Zawahałam się gdyż AbstractPlotterAdapter nie implementuje interfejsu klienta. To jak to w końcu jest? Ten warunek musi być spełniony?

Reply:
Źle przeczytałem o "nieimplementowaniu" interfejsu klienta, ale ostatecznie i tak się nie zgadzam. AbstractPlotter implementuje interfejs klienta, więc AbstractPlotterAdapter też. 
Implementacja jest faktycznie zgodna z duchem dekoratora,, ale pytanie jest nadal otwarte. 
Przyznam też, że dokładnie nie obejrzałem kodu, ale AbstractPlotterAdapter nie działa poprawnie (wydaje mi się, że słabo go pani przetestowała).

Reply:
Musiałabym wiedzieć co znaczy, że nie działa poprawnie :) Tzn. test Jane się wyświetla (jako domek), jeśli ma się wyświetlić coś innego to myślę, że też w pdf'ie przydałaby się wzmianka o tym jakiego efektu oczekujemy.

Reply:
Myślę i myślę i nie wiem jak Jane naprawić, jeśli inaczej zbudować AbstractPlotterAdapter to też nie mam pomysłu jak :(
A jeszcze wracając do Pana sugestii o przeciążeniu setPosition w tym adapterze to się nie da bo w AbstractPlotter jest to klasa finalna.